### ISSUE TITLE
How to deal with controlled terms?

### ISSUE THREAD

**Issue by Kleinjohann** [openMetadataInitiative/openMINDS_Python#24]
Hi!
I'm currently working on extracting openMINDS core metadata from existing metadata collections, and I'm using openMINDS_Python to automatise this.

I got stuck trying to set properties for which a library of instances is already provided by openMINDS, e.g. `data_type` of `DatasetVersion`. Is there a best practice for this? Do I instantiate a `SemanticDataType` with the correct `id` and `name` myself or is there a way to auto-generate it? And can I somehow retrieve and search a library of predefined terms?

Sorry if I'm missing something obvious here, I'm hoping this can be resolved faster and easier by asking than by poking around in the code.
Maybe a usage example that involves a controlled term could help to get people started on this?

**Comment by lzehl** [comment]
I think the status is the following: Through the Python library you should be able to make use of controlled terms but you should not have to (meaning you can also create your own terms for the different terminology types). I'm not sure though in how far the controlled terms of openMINDS are integrated yet as selection in the Python module. Potentially this is currently not the case meaning you would need to hard code this right now (collecting the instances IDs of the controlled terms you want to use from the documentation).

@apdavison can answer this question better though, and let you know what is planned or working already.

**Comment by lzehl** [comment]
@Kleinjohann we just merged a PR that should help you. Documentation will follow but please find below a small example on how you can retrieve openMINDS instances now.


```python
from openminds.latest.controlled_terms import AgeCategory, MolecularEntity

# instances can be called as attributes of their type
AgeCategory.adult 

# instances can also be called by name (which includes synonyms)
MolecularEntity.by_name("HEPES") == MolecularEntity.by_name("4-(2-hydroxyethyl)-1-piperazine ethanesulfonic acid")
```

**Comment by apdavison** [comment]
Controlled terms are now available as class attributes, as of release 0.2.0
